src/main/webapp/js/instructorFeedbackResults.js

```diff
@@ -205,40 +205,82 @@ function updateStatsCheckBox() {
+    var COLLAPSE = 'Collapse';
+    // {@code panels} is undefined when the method is called by clicking on collapse panels button.
+    var panelsDefined = panels || $('div.panel-collapse');
+    var isElementAnExpandButton = $(element).html().trim().startsWith(EXPAND);
+
+    if (isElementAnExpandButton) {
+        expand(panelsDefined, element);
+        replaceInHtml(element, EXPAND, COLLAPSE);
     } else {
-        var j = 0;
-        for (var k = 0; k < panels.length; k++) {
-            if ($(panels[k]).attr('class').indexOf('in') !== -1) {
-                setTimeout(hideSingleCollapse, 100 * j, panels[k]);
-                j++;
+        collapse(panelsDefined);
+        replaceInHtml(element, COLLAPSE, EXPAND);
+    }
+}
+/**
+ * Expand each panel. If the panel data is not loaded, load it by Ajax first before expansion.
+ */
+function expand(panels, element) {
+    var isElementCollapsePanelsButton = $(element).is($('#collapse-panels-button'));
+    // Ajax class is only found in {@code panels} if the element is collapse panels button.
+    var mightNeedAjaxLoading = isElementCollapsePanelsButton;
+    var i = 0;
+    for (var idx = 0; idx < panels.length; idx++) {
+        if (mightNeedAjaxLoading) {
+            // When the panel's parent element has class ajax_auto or ajax-response-auto, the panel data has
+            // not been loaded yet. We will need to load the panel data by Ajax.
+            var $ajaxAuto = $(panels[idx]).parent().children('.ajax_auto');
+            var $ajaxResponseAuto = $(panels[idx]).parent().children('.ajax-response-auto');
+            var hasAjaxAutoLoading = $ajaxAuto.length !== 0;
+            var hasAjaxResponseAutoLoading = $ajaxResponseAuto.length !== 0;
+
+            // Trigger Ajax loading by clicking on the element. When clicked, the panel data is loaded,
+            // and ajax_auto or ajax-response-auto class will be removed from the element.
+            if (hasAjaxAutoLoading) {
+                $ajaxAuto.click();
+            }
+            if (hasAjaxResponseAutoLoading) {
+                $ajaxResponseAuto.click();
+            }
+
+            var elementHasAjaxAndIsClicked = hasAjaxAutoLoading || hasAjaxResponseAutoLoading;
+            if (elementHasAjaxAndIsClicked) {
+                // When the element has Ajax class, its panel is in collapsed state. Clicking will expand
+                // its panels.
+                continue;
             }
         }
-        var htmlStr = $(e).html();
-        htmlStr = htmlStr.replace(collapse, expand);
-        $(e).html(htmlStr);
-        var tooltipStr = $(e).attr('data-original-title').replace(collapse, expand);
-        $(e).attr('title', tooltipStr).tooltip('fixTitle').tooltip('show');
+
+        if ($(panels[idx]).attr('class').indexOf('in') === -1) {
+            // Expand the panel if the panel is collapsed at the moment
+            setTimeout(showSingleCollapse, 50 * i, panels[idx]);
+            i++;
+        }
```

Comment:
Is it possible to extract a method that expands only one panel and call that method in a loop?